research/src/pages/radio-button/radio-button.proposal.mdx

```diff
@@ -0,0 +1,185 @@
+    <div part="checked-indicator"></div>
+  </slot>
+</oui-radio-button>
+```
+
+#### Slots
+
+| Slot Name           | Description                                                | Fallback Content                                 |
+| ------------------- | ---------------------------------------------------------- | ------------------------------------------------ |
+| `label`             | Add custom markup for the control's label                  | Empty                                            |
+| `checked-indicator` | Content to indicate the radio button is in a checked state | Element with checked and indeterminate indcators |
+
+#### CSS Parts
+
+| Slot Name           | Description                                                   |
+| ------------------- | ------------------------------------------------------------- |
+| `label`             | The control's label (and implicit value)                      |
+| `checked-indicator` | Indicates the radio button is in a checked or unchecked state |
+
+## Radio Button Group Design
+
+This section relates to a `<oui-radio-button-group>`. It can host multiple `<oui-radio-button>` elements.
+
+### API
+
+#### Properties
+
+| Attribute Name | Type     | Default Value | Description                                                                                      |
+| -------------- | -------- | ------------- | ------------------------------------------------------------------------------------------------ |
+| `name`         | `string` | `null`        | Represents the name of the control, to be used upon form submission .                            |
+| `value`        | `string` | `null`        | The value of the selected choice.                                                                |
+| `autofocus`    | `bool`   | `false`       | Get focus by default. Defaults to the first option, unless explicitly defined among the options. |
+| `disabled`     | `bool`   | `false`       | Prevents the user from interacting with any of the choices. Defaults to false.                   |
+| `form`         | `string` | `null`        | Associates the element with a form in the document whose `id` is this value.                     |
+| `readonly`     | `bool`   | `false`       | Makes it unavailable for user interaction, but the selected choice should still be submitted.    |
+| `required`     | `bool`   | `false`       | Makes it required to have a selected choice.                                                     |
+
+#### Events
+
+...
+
+### Anatomy
+
+<RadioButtonGroupAnatomy />
+
+#### DOM Structure (Implicit Grouping)
+
+The following structure represents a group with the `choices` slot filled with radio buttons (their internal structures are omitted for brevity).
+
+```html
+<oui-radio-button-group name="group-name">
+  <slot name="label">
+    <label part="label"></label>
+  </slot>
+  <slot name="required-indicator">
+    <div part="required-indicator"></div>
+  </slot>
```

Comment:
This is interesting and something I hadn't thought of for other form elements which include labels and have a required attribute. @gregwhitworth we should look at adding this holistically where applicable.

Reply:
I won't take credit for this idea because I just took what was already being addressed for checkboxes: https://open-ui.org/components/checkbox.research.concepts 

However, I'm not sure if this is a good idea, if it's going to be only a visual indicator. Maybe if it could receive a custom text that would be read specifically to represent the "requiredness" of the input, then it would make more sense. What do you think?

Reply:
The use of the `label` element initially threw me off here as I assumed this was initially a discussion about the radio button, rather than the label (`legend`) for the radiogroup.

In either case though, this seems like something that would be nice to auto-include for people based on the requiredness of the group or individual form control.  If this markup translates to how this would actually render, being outside of the `label` / `legend`, it wouldn't be part of the accessible name of either - which is generally what would be desired as you don't want to have duplicate "required" announcements (if one were to put 'required' text into this slot).  If this indicator would be expected to be nested in the label/legend of the shadow dom of the component, e.g., 
```html
<fieldset role=radiogroup aria-required=true>
  <legend>[ label part ] [ required-indicator part ]</legend>
  ...
  or 
  <label for=x> [ label part ] [ required-indicator part ]</label>
  <input required id=x type=...>
</fieldset>
```
Then generally it _would_ be expected that the required indicator (particularly if it were text) be treated as visual only, so that the "required" state wasn't communicated twice as part of the state of the control AND in the accessible name of the control.

Reply:
I am not sure how this could be implemented in terms of accessibility, but maybe, if it is omitted, ATs could simply state that the field is required.

If it is included, however, and it has text attached to it, it could substitute the default AT required indication.

Reply:
the required attribute / aria-required=true attribute is what is important to expose to AT for the required nature of the grouping.  The indicator would be to visually inform what is then conveyed programmatically. 

>If it is included, however, and it has text attached to it, it could substitute the default AT required indication.

I'm not exactly sure what you're trying to say here.

Reply:
> If it is included, however, and it has text attached to it, it could substitute the default AT required indication.

Suppose an AT software might stop at a required field and say "Required" as it reads the `required` attribute. This would be what I expect to be default behavior.

Suppose then we have a `required-indicator` element on DOM, and it has a custom text saying "This field is required due to XYZ". Then, instead of simply reading "Required", it could read both or maybe only the text inside the `required-indicator`.

Just an idea.

Reply:
I will consider this to be resolved since I still abide by my proposal and I didn't hear back afterwards.